aws/resource_aws_cognito_identity_provider.go

```diff
@@ -25,13 +27,22 @@ func resourceAwsCognitoIdentityProvider() *schema.Resource {
 			"attribute_mapping": {
 				Type:     schema.TypeMap,
 				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringLenBetween(1, 32),
+				},
 			},
 
 			"idp_identifiers": {
 				Type:     schema.TypeList,
 				Optional: true,
```

Comment:
While we're updating here, seems like this should include `MaxItems: 50` according to the [API Reference](https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_CreateIdentityProvider.html)

```suggestion
				Optional: true,
				MaxItems: 50,
```